### PR TITLE
Feature/apps 1772 upgrade to log4jv2

### PR DESCRIPTION
**Pull request by kavitshah-gl**
These changes are for the upgrade to Log4jv2.

### CODE REVIEW

File: samples/consuming-tas-utility/src/test/resources/log4j2.properties [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+###### Console appender definition #######
+# All outputs currently set to be a ConsoleAppender.
+# Direct log messages to stdout
+appender.ConsoleAppender.type=Console
+appender.ConsoleAppender.name=ConsoleAppender
+appender.ConsoleAppender.layout.type=PatternLayout
+# use log4j NDC to replace %x with tenant domain / username
+appender.ConsoleAppender.layout.pattern=%d{ISO8601} %x %-5p [%c{3}] [%t] %replace{%m}{[\r\n]+}{}%n
+#log4j.appender.Console.layout.ConversionPattern=%d{ABSOLUTE} %-5p [%c] %m%n
```
Comment (dsibilio) [inline samples/consuming-tas-utility/src/test/resources/log4j2.properties]: We should remove commented out configuration with Log4j 1.x syntax.

File: samples/consuming-tas-utility/src/test/resources/log4j2.properties [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Direct log messages to a log file
+appender.RollingAppender.type=RollingFile
+appender.RollingAppender.name=RollingAppender
+appender.RollingAppender.fileName=./target/reports/alfresco-tas.log
+appender.RollingAppender.filePattern=./target/reports/alfresco-tas.log.%d{yyyy-MM-dd}
+appender.RollingAppender.layout.type=PatternLayout
+appender.RollingAppender.layout.pattern=%d{yyyy-MM-dd} %d{ABSOLUTE} %-5p [%c] [%t] %replace{%m}{[\r\n]+}{}%n
+appender.RollingAppender.policies.type = Policies
+appender.RollingAppender.policies.time.type = TimeBasedTriggeringPolicy
+appender.RollingAppender.policies.time.interval = 1
```
Comment (dsibilio) [inline samples/consuming-tas-utility/src/test/resources/log4j2.properties]: In the original file the rolling strategy was not based on time, but rather on size combined with a max backup count similar to this:

```
###### File appender definition #######
appender.rolling.type=RollingFile
appender.rolling.name=RollingAppender
appender.rolling.fileName=./target/reports/alfresco-tas.log
appender.rolling.filePattern=./target/reports/alfresco-tas.log.%i
appender.rolling.layout.type=PatternLayout
appender.rolling.layout.pattern=[%t] %d{HH:mm:ss} %-5p %c{1}:%L - %replace{%m}{[\r\n]+}{}%n
appender.rolling.policies.type=Policies
appender.rolling.policies.size.type=SizeBasedTriggeringPolicy
appender.rolling.policies.size.size=10MB
appender.rolling.strategy.type=DefaultRolloverStrategy
appender.rolling.strategy.max=10
```

File: samples/consuming-tas-utility/src/test/resources/log4j2.properties [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+rootLogger.appenderRef.rolling.ref=RollingAppender
+
+###### Console appender definition #######
+# All outputs currently set to be a ConsoleAppender.
+# Direct log messages to stdout
+appender.ConsoleAppender.type=Console
+appender.ConsoleAppender.name=ConsoleAppender
+appender.ConsoleAppender.layout.type=PatternLayout
+# use log4j NDC to replace %x with tenant domain / username
+appender.ConsoleAppender.layout.pattern=%d{ISO8601} %x %-5p [%c{3}] [%t] %replace{%m}{[\r\n]+}{}%n
```
Comment (dsibilio) [inline samples/consuming-tas-utility/src/test/resources/log4j2.properties]: Is there a reason why we're changing the layout pattern compared to the original implementation? 

In general I'd expect that we try and keep the patterns that are already present, e.g. in this case: `[%t] %d{HH:mm:ss} %-5p %c{1}:%L - %m%n` whilst replacing the newlines therefore making it: `[%t] %d{HH:mm:ss} %-5p %c{1}:%L - %replace{%m}{[\r\n]+}{}%n`

**Note: this applies to all patterns in the PR.**

File: samples/consuming-tas-utility/src/test/resources/log4j2.properties [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Set root logger level to error
+#log4j.rootLogger=${app.log.root.level}, Console, File
```
Comment (dsibilio) [inline samples/consuming-tas-utility/src/test/resources/log4j2.properties]: We should remove commented out configuration with Log4j 1.x syntax.

File: samples/consuming-tas-utility/src/test/resources/log4j2.properties [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+# TestRail particular log file
+# Direct log messages to a log file
+appender.TestRollingAppender.type=RollingFile
+appender.TestRollingAppender.name=TestRollingAppender
+appender.TestRollingAppender.fileName=./target/reports/alfresco-testrail.log
+appender.TestRollingAppender.layout.type=PatternLayout
+appender.TestRollingAppender.layout.pattern=%d{HH:mm:ss} %-5p %c{1}:%L - %m%n
+appender.TestRollingAppender.strategy.type = DefaultRolloverStrategy
+appender.TestRollingAppender.strategy.max = 10
```
Comment (dsibilio) [inline samples/consuming-tas-utility/src/test/resources/log4j2.properties]: This configuration should be largely similar to the one I just proposed earlier as a literal translation of the previous implementation, with minor differences in the file name, appender name, layout pattern structure.

File: src/test/resources/log4j2.properties [hunk 6]
```diff
@@ -0,0 +1,37 @@
+# Set root logger level to error
+#log4j.rootLogger=${app.log.root.level}, Console, File
```
Comment (dsibilio) [inline src/test/resources/log4j2.properties]: Should be removed.

File: src/test/resources/log4j2.properties [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+###### Console appender definition #######
+# All outputs currently set to be a ConsoleAppender.
+# Direct log messages to stdout
+appender.console.type=Console
+appender.console.name=ConsoleAppender
+appender.console.layout.type=PatternLayout
+# use log4j NDC to replace %x with tenant domain / username
+appender.console.layout.pattern=%d{ISO8601} %x %-5p [%c{3}] [%t] %replace{%m}{[\r\n]+}{}%n
+#log4j.appender.Console.layout.ConversionPattern=%d{ABSOLUTE} %-5p [%c] %m%n
```
Comment (dsibilio) [inline src/test/resources/log4j2.properties]: Should be removed.

File: src/test/resources/log4j2.properties [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+# TestRail particular log file
+# Direct log messages to a log file
+appender.TestRollingAppender.type=RollingFile
+appender.TestRollingAppender.name=TestRollingAppender
+appender.TestRollingAppender.fileName=./target/reports/alfresco-testrail.log
+appender.TestRollingAppender.layout.type=PatternLayout
+appender.TestRollingAppender.layout.pattern=%d{HH:mm:ss} %-5p %c{1}:%L - %m%n
+appender.TestRollingAppender.strategy.type = DefaultRolloverStrategy
+appender.TestRollingAppender.strategy.max = 10
```
Comment (dsibilio) [inline src/test/resources/log4j2.properties]: Pretty much same as the previously highlighted configuration translation issue, I believe we weren't faithful in the translation of the configuration and I'm not sure if this was a desired change or not.

**Review by dsibilio — CHANGES_REQUESTED** [review]

**Review by dsibilio — APPROVED** [review]

**Review by MohinishSah — APPROVED** [review]
LGTM

**Review by damianujma — APPROVED** [review]

**Review by aonych — APPROVED** [review]

**Pull request merged**